File: contentservice/src/main/java/hu/martontoth/notflix/controller/ContentController.java
package hu.martontoth.notflix.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.core.env.Environment;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * Created by Márton Tóth
 */
@RestController
public class ContentController {

    @Autowired
    private Environment environment;

    @Value("${stuff}")
    private String stuff;

    @RequestMapping("/")
    public String example() {
        return "Ahoy from port " + environment.getProperty("local.server.port") + "!\nCurrent stuff value is " + stuff;
    }

}

File: content/src/main/kotlin/hu/martontoth/content/content/ContentController.kt
package hu.martontoth.content.content

import org.springframework.beans.factory.annotation.Value
import org.springframework.core.io.FileSystemResource
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
@RequestMapping("api/content")
class ContentController(
        @Value("\${content.path}")
        private val contentPath: String?
) {

    @GetMapping("{id}/thumbnail")
    fun thumbnail(@PathVariable("id") id: Long): ResponseEntity<Mono<FileSystemResource>> {
        val fileSystemResource = FileSystemResource("$contentPath/$id/thumbnail.jpg")
        return if (fileSystemResource.exists()) {
            ResponseEntity.ok()
                    .header("Content-Type", "image/jpeg")
                    .body(Mono.just(fileSystemResource))
        } else {
            ResponseEntity.notFound().build()
        }
    }

    @GetMapping("{id}/stream")
    fun stream(@PathVariable("id") id: Long): ResponseEntity<Mono<FileSystemResource>> {
        val fileSystemResource = FileSystemResource("$contentPath/$id/stream.m4v")
        return if (fileSystemResource.exists()) {
            ResponseEntity.ok()
                    .header("Content-Type", "video/mp4")
                    .body(Mono.just(fileSystemResource))
        } else {
            ResponseEntity.notFound().build()
        }
    }
}